nemo/collections/nlp/models/text_classification_model.py

```diff
@@ -132,40 +172,44 @@ def validation_epoch_end(self, outputs):
         """
         avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
         # val_acc = sum([x['n_correct_pred'] for x in outputs]) / sum(x['n_pred'] for x in outputs)
-        # tensorboard_logs = {'val_loss': avg_loss, 'val_acc': val_acc}
-        return {'val_loss': avg_loss}  # , 'log': tensorboard_logs}
+        tensorboard_logs = {'val_loss': avg_loss}  # , 'val_acc': val_acc}
+        # return {'val_loss': avg_loss}  # , 'log': tensorboard_logs}
+        return {'val_loss': avg_loss, 'log': tensorboard_logs}
 
-    def setup_training_data(self, file_path, dataloader_params={}):
-        self._train_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_training_data(self, train_data_config: Optional[DictConfig]):
+        # if 'shuffle' not in train_data_config:
+        #     train_data_config['shuffle'] = True
+        self._train_dl = self.__setup_dataloader_from_config(cfg=train_data_config)
 
-    def setup_validation_data(self, file_path, dataloader_params={}):
-        self._validation_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_validation_data(self, val_data_config: Optional[DictConfig]):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg=val_data_config)
 
-    def setup_test_data(self, file_path, dataloader_params={}):
-        self._test_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_test_data(self, test_data_config: Optional[DictConfig]):
+        self._test_dl = self.__setup_dataloader(cfg=test_data_config)
 
-    def __setup_dataloader(
-        self, input_file, dataloader_params={},
-    ):
+    def __setup_dataloader_from_config(self, cfg: TextClassificationDataConfig):
+        input_file = f'{self.data_dir}/{cfg.prefix}.tsv'
+        # TODO: check that file exists
         dataset = TextClassificationDataset(
             input_file=input_file,
             tokenizer=self.tokenizer,
-            max_seq_length=dataloader_params.get("max_seq_length", 512),
-            num_samples=dataloader_params.get("num_samples", -1),
-            shuffle=dataloader_params.get("shuffle", False),
-            use_cache=dataloader_params.get("use_cache", False),
+            max_seq_length=self.max_seq_length,
+            num_samples=cfg.num_samples,
+            shuffle=cfg.shuffle,
+            use_cache=cfg.use_cache,
         )
 
         return torch.utils.data.DataLoader(
             dataset=dataset,
-            batch_size=dataloader_params.get("batch_size", 64),
-            shuffle=dataloader_params.get("shuffle", False),
-            num_workers=dataloader_params.get("num_workers", 0),
-            pin_memory=dataloader_params.get("pin_memory", False),
+            batch_size=cfg.batch_size,
```

Comment:
Maybe you want to use the defaults here, pin memory is pretty niche to supply in a config ? Maybe I'm being pedantic.

Reply:
Agreed. I'm hoping we can make the config.yaml to be used more as a quick reference of things that are commonly changed. I love looking at a new model and being able to skim the yaml and get a good idea of what's going on.